src/admin/components/forms/field-types/Array/index.tsx

```diff
@@ -282,7 +284,14 @@ const ArrayFieldType: React.FC<Props> = (props) => {
                           className={`${baseClass}__row`}
                           key={row.id}
                           dragHandleProps={providedDrag.dragHandleProps}
-                          header={`${labels.singular} ${rowNumber >= 10 ? rowNumber : `0${rowNumber}`}`}
+                          header={(
+                            <RowLabel
```

Comment:
It's a bit confusing to me that the prop name is called `header` but the built-in component is called `RowLabel`. We should flatten and either call everything `RowLabel` or `RowHeader`. Then the prop for the row component would be `header` or `label`, but it would match.